### ISSUE TITLE
Write queries with missing parameters fail silently

### ISSUE THREAD

**Issue by hemberger** [staabm/phpstan-dba#503]
I think I've isolated an issue where queries with errors are not being reported. If a write query is missing a parameter, then it seems that no errors are reported for that query under any circumstances.

I'm using a custom prepared statement method (using PDO under the hood) with the following configuration:
```neon
class: staabm\PHPStanDba\Rules\SyntaxErrorInPreparedStatementMethodRule
tags: [phpstan.rules.rule]
arguments:
    classMethods:
        - 'My\Database::write'
```

The most generic example I can find is the following:
```php
$db->write('DELETE FROM my_table WHERE col1 = :col1 AND col2 = :col2', [
    'col1' => $value,
]);
```
This reports no error, even though the `col2` placeholder is missing. Furthermore, I can introduce an error into the query string by, e.g., using the wrong table name:
```php
$db->write('DELETE FROM table_does_not_exist WHERE col1 = :col1 AND col2 = :col2', [
    'col1' => $value,
]);
```
This will also report no error.

In either case, if I change `DELETE` to `SELECT`, all errors are reported:

```php
$db->write('SELECT FROM my_table WHERE col1 = :col1 AND col2 = :col2', [
    'col1' => $value,
]);
```
> Query expects placeholder :col2, but it is missing from values given.

```php
$db->write('SELECT FROM table_does_not_exist WHERE col1 = :col1 AND col2 = :col2', [
    'col1' => $value,
]);
```
> Query error: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'dbname.table_does_not_exist' doesn't exist (42S02).

Turning on debug mode (`$config->debugMode(true);`) doesn't provide any additional output.

If there is any additional information I can provide, please don't hesitate to let me know. Thanks for your time!

**Comment by staabm** [comment]
thx for the awesoeme report, including examples.

Support for writable queries is quite recent and was added with https://github.com/staabm/phpstan-dba/pull/483
it would be great to also have the placeholder validation for the writable queries.

would be great you could have a look into it
